src/platforms/f3/platform.c

```diff
@@ -112,9 +118,71 @@ bool platform_nrst_get_val(void)
 	return (gpio_get(NRST_PORT, NRST_PIN)) ? false : true;
 }
 
+
+static void adc_init(void)
+{
+
+	//ADC
+	rcc_periph_clock_enable(RCC_ADC12);
+	//ADC
+	gpio_mode_setup(VTREF_PORT, GPIO_MODE_ANALOG, GPIO_PUPD_NONE, VTREF_PIN);
+	//gpio_mode_setup(GPIOA, GPIO_MODE_ANALOG, GPIO_PUPD_NONE, GPIO1);
+	adc_power_off(ADC1);
+	adc_set_clk_prescale(ADC1, ADC_CCR_CKMODE_DIV2);
+	adc_set_single_conversion_mode(ADC1);
+	adc_disable_external_trigger_regular(ADC1);
+	adc_set_right_aligned(ADC1);
+	/* We want to read the temperature sensor, so we have to enable it. */
+	//adc_enable_temperature_sensor();
+	adc_set_sample_time_on_all_channels(ADC1, ADC_SMPR_SMP_601DOT5CYC);
+	uint8_t channel_array[] = { 1 }; /* ADC1_IN1 (PA0) */
+	adc_calibrate(ADC1);
+	adc_set_regular_sequence(ADC1, 1, channel_array);
+	adc_set_resolution(ADC1, ADC_CFGR1_RES_12_BIT);
+	adc_power_on(ADC1);
+
+	/* Wait for ADC starting up. */
+	int i;
+	for (i = 0; i < 800000; i++)
+		__asm__("nop");
+	
+
+
+}
+
+
+uint32_t platform_target_voltage_sense(void)
+{
+	/*
+	 * Returns the voltage in tenths of a volt (so 33 means 3.3V)
+	 */
+	adc_start_conversion_regular(ADC1);
+	while (!(adc_eoc(ADC1)))
+		continue;
+	uint32_t val=adc_read_regular(ADC1);
+	return (val * 99U) / 8191U;;
+	//return val;
+	
+	return 0;
```

Comment:
This is unreachable, please remove it, along with the commented out `return val;`